docker_images/java/wrapper/src/main/java/glue/ServiceGlue.java

```diff
@@ -99,10 +104,21 @@ private void invokeMethodCommon(String connectionId, String deviceId, String mod
             }
             System.out.printf("invoke returned%n");
             System.out.println(result);
-            handler.handle(Future.succeededFuture(result));
+            handler.handle(Future.succeededFuture(makeMethodResultThatEncodesCorrectly(result)));
```

Comment:
Our JSON encoder seems not a fan of Gson, so taking the workaround here by referring to the same approach in `ModuleGlue` class.